src/modules/file/entities/file.entity.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { Exclude } from 'class-transformer';
+
+import { User } from '../../user/entities/user.entity';
+
+export enum ScanResult {
+  MALICIOUS = 'MALICIOUS',
+  PASSED = 'PASSED',
+  NOSCAN = 'NOSCAN',
+}
+
+export type S3Status = 'in_progress' | 'finished';
+
+@Entity('files')
+export class File {
+  @PrimaryGeneratedColumn()
+    id: number;
+
+  @Column()
+    filename: string;
+
+  @Column()
+    extension: string;
+
+  @Column()
+    content_type: string;
+
+  @Column()
+  @Exclude({ toPlainOnly: true })
+    s3_path: string;
+
+  @Column({ default: 'in_progress' })
+    s3_status: S3Status;
+
+  @Column()
+    filesize: string;
+
+  @Column()
+    chunk_count: number;
+
+  @Column()
+    chunk_size: number;
+
+  @ManyToOne(() => User, (user) => user.id)
+  @JoinColumn()
+    user: User;
+
+  @Column()
+    user_id: number;
+
+  @Column({ nullable: true })
+    download_count: number;
+
+  @Column({ type: 'timestamp', nullable: true })
+    last_download_at: Date;
+
+  @Column({ nullable: true })
+    max_download_count: number;
+
+  @CreateDateColumn({
```

Comment:
It's better to declare created_at and updated_at at the end of model, since they are present in almost any model